proposed/2022/PrivateAssetIndepdentFromExcludeAsset.md

```diff
@@ -0,0 +1,137 @@
+# Opt-in `PrivateAssets` asset flow control option precedence over `IncludeAssets/ExcludeAssets` option
+
+- Author Name <https://github.com/erdembayar>
+- Start Date 2022-12-07
+- GitHub Issue <https://github.com/NuGet/Home/issues/6938>
+- GitHub PR <https://github.com/NuGet/NuGet.Client/pull/4976>
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+Currently `IncludeAssets/ExcludeAssets` options take precedence over `PrivateAssets` asset control option when creating package, so in many cases `IncludeAssets/ExcludeAssets` options completely eclipse the `PrivateAssets` option so doesn't let assets flow to consuming parent project. This proposal introduces new a `PackPrivateAssetsFlow` opt-in property so it let `PrivateAssets` asset control option takes precedence over `IncludeAssets/ExcludeAssets` options to enable asset flow to consuming parent projects.
+
+## Motivation
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+A package author should be able control which asset flow to parent consuming projects when this project is consumed as package. It enables parent projects to consume transitive packages without having to directly reference them, therefore it reduces the number of packages developers need to keep track of.
+We couldn't make this default experience because it'll break customers who rely on current behavior.
```

Comment:
Have we determined that the existing behavior is in fact deterministic? My experiments suggest that it is perhaps not. If it isn't deterministic, then IMO it's broken and maybe we should fix that rather than add an opt-in property for getting deterministic behavior.

Reply:
I tested what happens if we make this default experience, then [35-unit tests broken] (https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=7058885&view=ms.vss-test-web.build-test-results-tab). I believe it's quite a high number, so hesitate to make it default. 
Also, I mentioned in table `Functional explanation section` it might break compile/runtime scenarios. Tbh I don't know how widespread it's that scenarios I mentioned there. 
How about we make it opt-in for now and create a follow up issue for making it default experience. If a follow up issue gathers enough votes, then we can make it default experience that time.
Alternative to this approach would be make this default experience and add Environment var for fall back to previous experience in case there’s regression for some customers.